tests/unit/test_controller.py

```diff
@@ -0,0 +1,41 @@
+import asynctest
+
+
+class TestControllerConnect(asynctest.TestCase):
+    @asynctest.patch('juju.client.connector.Connector.connect_controller')
+    async def test_controller_connect_no_args(self, mock_connect_controller):
+        from juju.controller import Controller
```

Comment:
Why importing this here rather than at module level?

Reply:
I'm following the pattern used in the other tests.

Reply:
I imagine this pattern started from confusion about how to get something mocked properly.  There are better ways to address it but it's not a big deal.  There are some other test fixes that I've been meaning to address and I can clean these up at that time.